### ISSUE TITLE
tsx 4.13.1 fails to find exported module in @aws-sdk/client-iot-data-plane with Node.js 20

### ISSUE THREAD

**Issue by coderbyheart** [privatenumber/tsx#577]
### Acknowledgements

- [X] I read the documentation and searched existing issues to avoid duplicates
- [X] I am reporting a confirmed bug (not seek debugging help)
- [X] I understand this is a collaborative project and relies on my contributions
- [X] I read and understood the [Contribution guide](https://github.com/privatenumber/tsx/blob/develop/CONTRIBUTING.md)

### Minimal reproduction URL

https://github.com/coderbyheart/tsx-import-bug/actions/runs/9414431871/job/25933332684#step:5:6

### Problem & expected behavior (under 200 words)

This works with 4.12.1: https://github.com/coderbyheart/tsx-import-bug/actions/runs/9414522107/job/25933611074#step:6:5

## package.json

```json
{
  "devDependencies": {
    "@aws-sdk/client-iot-data-plane": "3.592.0",
    "tsx": "4.13.1"
  },
  "engines": { "node": ">=20" }
}
```

## index.ts

```typescript
import { UpdateThingShadowCommand } from '@aws-sdk/client-iot-data-plane';

new UpdateThingShadowCommand({} as any);
```

## Output

``` bash
$ npx tsx -v
tsx v4.13.1
node v20.14.0

$ npx tsx index.ts

import { UpdateThingShadowCommand } from "@aws-sdk/client-iot-data-plane";
^

SyntaxError: The requested module '@aws-sdk/client-iot-data-plane' does not provide an export named 'UpdateThingShadowCommand'
at ModuleJob._instantiate (node:internal/modules/esm/module_job:134:21)
at async ModuleJob.run (node:internal/modules/esm/module_job:217:5)
at async ModuleLoader.import (node:internal/modules/esm/loader:316:24)
at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:123:5)

Node.js v20.14.0
```

### Bugs are expected to be fixed by those affected by it

- [X] I'm interested in working on this issue

### Compensating engineering work financially will speed up resolution

- [ ] I'm willing to offer financial support

**Comment by m4rvr** [comment]
I have the same issue with the `envalid` npm package (https://www.npmjs.com/package/envalid)

**Comment by AliMirlou** [comment]
I'm facing the same issue on most installed packages.

This commit has caused the issue: [https://github.com/privatenumber/tsx/commit/0a78bfd11b20fcb6460fdcdea7d3b1daebfe3ad0](https://github.com/privatenumber/tsx/commit/0a78bfd11b20fcb6460fdcdea7d3b1daebfe3ad0)

**Comment by karlhorky** [comment]
I created another reproduction in my issue also about `The requested module '...' does not provide an export named '...'`, reported over here:

- https://github.com/privatenumber/tsx/issues/578

**Comment by privatenumber** [comment]
:tada: This issue has been resolved in v4.13.2

If you appreciate this project, please consider [supporting this project by sponsoring](https://github.com/sponsors/privatenumber) :heart: :pray:

**Comment by privatenumber** [comment]
Thanks for the issue @coderbyheart! Should be fixed now ^
